backend/reservations/views/admin.py

```diff
@@ -142,144 +144,112 @@ def guest_search(arr, low, high, x):
+            guest_rows.append(stripd)
 
     characters = string.ascii_letters + string.digits + string.punctuation
-
-    for guest_new in dr:
-        logger.debug(f'new guest: {guest_new}')
-        guest_entries = Guest.objects.filter(email=guest_new["email"])
-        if ("Directed Sale Room (1 King Bed)" in guest_new["product"]):
-            guest_new["product"] = "King"
-        if ("Directed Sale Room (1 Queen Bed)" in guest_new["product"]):
-            guest_new["product"] = "Queen"
-
-        if ("Hard Rock" in guest_new["product"]):
+    for guest_new in guest_rows:
+        if(guest_new['product'][:3] == "Art"):
             continue
-
-        # If email doesnt exist, send email and create guest
-        if (guest_search(gl, 0, glen, guest_new["email"])==None and len(guest_entries)==0):
-            logger.info(f'Email doesnt exist. Creating new guest contact.')
-            otp = ''.join(random.choice(characters) for i in range(10))
-            guest_contact_new(guest_new, otp)
+        if ("Hard Rock" in guest_new["product"]):
             continue
+        guest_entries = Guest.objects.filter(email=guest_new["email"])
+        trans_code = guest_new['transferred_from_code']
+        tix_exist = [guest.ticket for guest in guest_entries if guest.ticket==guest_new['ticket_code']]
 
-        # If email does exist. check whether ticket exists. if not, create guest
-        if(len(Guest.objects.filter(ticket=guest_new["ticket_code"]))==0):
+        # Create with email
+        if(len(guest_entries)==0):
+            logger.info(f'Email doesnt exist: {guest_new["email"]} ql: {guest_entries}. Creating new guest contact.')
             otp = ''.join(random.choice(characters) for i in range(10))
-            guest_contact_exists(guest_new, otp)
-
-        # If email does exist. check whether ticket exists. if so, update the guest entry.
-        # This case if for resigning room/tickets to new owner. update operation so secret party export is source of authority.
-        elif(len(Guest.objects.filter(ticket=guest_new["ticket_code"]))!=0):
+            guest_contact_new(guest_new, otp, email_onboarding=True)
+        # Update from ticket transfer
+        elif(trans_code!=""):
+            try:
+                existing_guest = Guest.objects.filter(ticket=trans_code)[0]
+            except IndexError as e:
+                logger.warn(f'[-] Ticket transfer but no previous ticket id found')
+                continue
+            existing_room = existing_ticket.room
+            logger.info(f'Ticket is a transfer. ')
             otp = ''.join(random.choice(characters) for i in range(10))
-            guest_contact_update(guest_new, otp)
+            if(len(guest_entries)==0):
+                guest_contact_new(guest_new, otp, email_onboarding=True, room=existing_room)
+            else:
+                guest_contact_new(guest_new, otp, email_onboarding=False, room=existing_room)
+            existing_guest.delete()
```

Comment:
is there a reason we cannot modify the guest object / entry?

Reply:
Also if we are generating a new otp and not sending them an email won't that confuse things?

Reply:
I delete so it gets a new record id in case the old one is lingering anywhere and tries to point at something that changed. With a project this small could prolly just update as suggested.

Reply:
Otp isn't actually needed in any entry except the first one created. Just needs a val. Could update the models to allow null. Would prolly b safer

Reply:
Can we define "lingering"? If you update, the reference (i.e. to room or admin) doesn't change at all? I'll definitely be updating the models to have some more `null`s

Reply:
I had been trying to use guest ID on the frontend as a uid and found issues with them not existing between updats or logins, donnt recall. Not doing that currently to so maybe fine. i look

Reply:
Hmmm I think saves this change for next PR
if we make an update function it would be nice to separate out the onboarding from the create function. so we can use onboarding in all cases. moving that stuff around would block u longer.

Reply:
aite i might take a poke at it in my pr given the schema changes and already needing to be able to deal with guest updates